packages/@react-aria/utils/src/useId.ts

```diff
@@ -26,9 +26,12 @@ export let idsUpdaterMap: Map<string, { current: string | null }[]> = new Map();
 // This allows us to clean up the idsUpdaterMap when the id is no longer used.
 // Map is a strong reference, so unused ids wouldn't be cleaned up otherwise.
 // This can happen in suspended components where mount/unmount is not called.
-let registry = new FinalizationRegistry<string>((heldValue) => {
-  idsUpdaterMap.delete(heldValue);
-});
+let registry;
+if (typeof window !== 'undefined' && window.FinalizationRegistry) {
```

Comment:
Do we need to have a window? Some server environments might also have FinalizationRegistry but not have a window. Any problem with doing this?

```suggestion
if (typeof FinalizationRegistry !== 'undefined') {
```